test/auth_aws/test_auth_aws.py

```diff
@@ -54,5 +54,40 @@ def test_connect_uri(self):
             client.get_database().test.find_one()
 
 
+class TestAWSLambdaExamples(unittest.TestCase):
+    def test_shared_client(self):
+        # Start AWS Lambda Example 1
+        from pymongo import MongoClient
+
+        client = MongoClient(host=os.environ["MONGODB_URI"])
+
+        def lambda_handler(event, context):
+            return client.db.command("ping")
+
+        # End AWS Lambda Example 1
+
+    @unittest.skipIf(
+        not os.environ.get("AWS_ACCESS_KEY_ID"), "Must have IAM credentials in " "environment."
+    )
+    def test_IAM_auth(self):
+        # Start AWS Lambda Example 2
+        import os
+
+        from pymongo import MongoClient
+
+        client = MongoClient(
+            host=os.environ["MONGODB_URI"],
+            username=os.environ["AWS_ACCESS_KEY_ID"],
+            password=os.environ["AWS_SECRET_ACCESS_KEY"],
```

Comment:
We don't event need `authSource`, since it defaults to `$external` per the [spec](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#obtaining-credentials)

Reply:
True but without authSource users will run into this error when authSource is defined in the URI:
```python
>>> MongoClient('mongodb://localhost/?authSource=admin', authMechanism='MONGODB-AWS')
Traceback (most recent call last):
...
  File "/Users/shane/git/mongo-python-driver/pymongo/auth.py", line 132, in _build_credentials_tuple
    "authentication source must be $external or None for MONGODB-AWS"
pymongo.errors.ConfigurationError: authentication source must be $external or None for MONGODB-AWS
```

Atlas sets authSource=admin in all SRV addresses by default.

Reply:
Understood, thanks

Reply:
https://jira.mongodb.org/browse/PYTHON-3331